File: app/src/main/java/com/zskjprojectj/andouclient/activity/WithdrawalActivity.java
package com.zskjprojectj.andouclient.activity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;

import com.zskjprojectj.andouclient.R;
import com.zskjprojectj.andouclient.base.BaseActivity;
import com.zskjprojectj.andouclient.base.BasePresenter;
import com.zskjprojectj.andouclient.http.ApiUtils;
import com.zskjprojectj.andouclient.http.BaseObserver;
import com.zskjprojectj.andouclient.http.HttpRxObservable;
import com.zskjprojectj.andouclient.utils.LoginInfoUtil;
import com.zskjprojectj.andouclient.utils.ToastUtil;

import java.io.IOException;

/**
 * 提现
 */
public class WithdrawalActivity extends BaseActivity {
    private Button btn_surewithdrawal;

    @Override
    protected void setRootView() {
        setContentView(R.layout.activity_withdrawal);
    }

    @Override
    protected void initData(Bundle savedInstanceState) {
        topView.setTitle("余额提现");
    }

    @Override
    protected void initViews() {
        btn_surewithdrawal = findViewById(R.id.btn_surewithdrawal);
        btn_surewithdrawal.setOnClickListener(view -> {
            EditText amountEdt = findViewById(R.id.amountEdt);
            float amount = Float.valueOf(amountEdt.getText().toString());
            if (amount <= 0) {
                ToastUtil.showToast("提现金额必须大于0！");
                return;
            }
            HttpRxObservable.getObservable(ApiUtils.getApiService().cash(
                    LoginInfoUtil.getUid(),
                    LoginInfoUtil.getToken(),
                    amount,
                    "13888888888",
                    "6222031300255596882"
            )).subscribe(new BaseObserver<Object>(mAt) {
                @Override
                public void onHandleSuccess(Object o) throws IOException {
                    jumpActivity(WithdrawalresultsActivity.class);
                }
            });
        });
    }

    @Override
    public void getDataFromServer() {

    }

    @Override
    protected BasePresenter createPresenter() {
        return null;
    }
}

File: app/src/main/java/com/zskjprojectj/andouclient/activity/WithdrawalActivity.kt
package com.zskjprojectj.andouclient.activity

import android.os.Bundle
import com.blankj.utilcode.util.ActivityUtils
import com.blankj.utilcode.util.RegexUtils
import com.blankj.utilcode.util.ToastUtils
import com.zhuosongkj.android.library.app.BaseActivity
import com.zhuosongkj.android.library.util.ActionBarUtil
import com.zhuosongkj.android.library.util.RequestUtil
import com.zskjprojectj.andouclient.R
import com.zskjprojectj.andouclient.http.ApiUtils
import com.zskjprojectj.andouclient.utils.LoginInfoUtil
import kotlinx.android.synthetic.main.activity_withdrawal.*

class WithdrawalActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ActionBarUtil.setTitle(mActivity, "余额提现")
        allBtn.setOnClickListener {
            amountEdt.setText(accountTxt.text.toString())
        }
        btn_surewithdrawal.setOnClickListener {
            val amount = amountEdt.text.toString().toFloatOrNull()
            if (amount == null || amount <= 0) {
                ToastUtils.showShort("请输入正确的提现金额")
            } else if (!RegexUtils.isMobileSimple(phoneEdt.text.toString())) {
                ToastUtils.showShort("请输入正确的手机号")
            } else if (accountEdt.text.toString().isEmpty()) {
                ToastUtils.showShort("清输入正确的银行账号")
            } else {
                RequestUtil.request(mActivity, true, false, {
                    ApiUtils.getApiService().cash(
                            LoginInfoUtil.getUid(),
                            LoginInfoUtil.getToken(),
                            amount,
                            phoneEdt.text.toString(),
                            accountEdt.text.toString()
                    )
                }, {
                    ActivityUtils.startActivity(WithdrawalresultsActivity::class.java)
                    finish()
                })
            }
        }
        RequestUtil.request(mActivity, true, true,
                {
                    ApiUtils.getApiService().cashDetail(
                            LoginInfoUtil.getUid(),
                            LoginInfoUtil.getToken(),
                            1
                    )
                },
                {
                    accountTxt.text = it.data.money
                })
    }

    override fun getContentView() = R.layout.activity_withdrawal
}